### PR TITLE
first version of algorithm learning and working +  finished vanilla policy gradient baseline implementation on four rooms

### PR DESCRIPTION
**Pull request by bohanwu**

### CODE REVIEW

File: policy_gradient.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                                                     name="avg_reward")
+        self.max_reward_placeholder = tf.placeholder(tf.float32, shape=(),
+                                                     name="max_reward")
+        self.std_reward_placeholder = tf.placeholder(tf.float32, shape=(),
+                                                     name="std_reward")
+
+        self.eval_reward_placeholder = tf.placeholder(tf.float32, shape=(),
+                                                      name="eval_reward")
+
+        tf.summary.scalar("Avg Reward", self.avg_reward_placeholder)
```
Comment (ByzanTine) [inline policy_gradient.py]: These names were warned by Tensorflow, consider rename them to 
Avg_Reward?

File: test_env/envs/fourrooms.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def inc(row, col, a):
                         newstate = to_s(newrow, newcol)
                         newletter = desc[newrow, newcol]
                         done = bytes(newletter) in b'GH'
-                        rew = float(newletter == b'G')
+                        rew = float(newletter == b'G') * 100 - 1
```
Comment (ByzanTine) [inline test_env/envs/fourrooms.py]: I would say -1 is not quite necessary if gamma = 0.9. 
But either way is fine.

**Comment by bohanwu** [comment]
Hey Jiayu, in code reviews I think the reviewer commits the pull request

**Comment by ByzanTine** [comment]
Hah, I don't have the permission:)

**Pull request merged**